pkg/steps/multi_stage.go

```diff
@@ -476,6 +476,7 @@ func (s *multiStageTestStep) generatePods(steps []api.LiteralTestStep, env []cor
 		pod.Labels[MultiStageTestLabel] = s.name
 		pod.Spec.ServiceAccountName = s.name
 		pod.Spec.TerminationGracePeriodSeconds = terminationGracePeriodSeconds
+		pod.Spec.DNSPolicy = coreapi.DNSPolicy(step.DNSPolicy)
```

Comment:
This needs to be conditional and not do anything if `step.DNSPolicy` is an empty string (please be sure to test this, it will break literally all jobs if this isn't correct)

Reply:
Also, can't we derive this? If a DNSServer is set, it should be `None`, otherwise we don't set it?